alacritty/src/display.rs

```diff
@@ -192,14 +202,21 @@ impl Display {
             wayland_event_queue = Some(display.create_event_queue());
         }
 
-        // Create the window where Alacritty will be displayed.
-        let size = dimensions.map(|(width, height)| PhysicalSize::new(width, height));
+        // Get the minimum size of the Alacritty window.
+        let min_size = window_size(
+            Dimensions::new(Column(MIN_COLS), Line(MIN_LINES)),
+            padding_x,
+            padding_y,
+            cell_width,
+            cell_height,
+        );
```

Comment:
Why would we spawn the window at the minimum size?

Reply:
We are not spawning it using the min size. Here we only calculate the min_size and pass it to the method that spawns the window so that we can set the min size (but not using it as the actual size) when spawning it.

Reply:
Why?

Reply:
I think I get a better idea about why you are asking why around the "always set min size" and "with min" behaviors.

The current implementation is like this:
```
1. spawn window with min size from estimated dpi 
2. calculate the min size from real dpi 
3. set min size to be the min size from real dpi
```
3 always happens, so 1 is not needed.

I think I should change it to:
```
1. spawn window with min size from estimated dpi 
2. calculate the min size from real dpi
if the min size from the real dpi is different from the min size from the estimated dpi:
   3. set min size to be the min size from real dpi
otherwise:
   4. do nothing
```

This way, the window is spawned with the correct min size from the beginning if we do the estimation correctly and we don't touch the min size again. I think this is semantically the behavior we want to have.

Reply:
We don't actually need to do any min size stuff unless the window is spawned with specific dimensions or the window is already spawned.

Reply:
In fact I'd say the whole min size thing has just completely distracted from the core of this PR and makes things more difficult rather than being useful tbh.

Reply:
So you're saying let's get rid of all the min size things?
Currently min size is calculated and used in renderer and window.
I'm not against removing min size things related to window (that is, calls to `set_min_inner_size` and `with_min_inner_size`) but I guess we should keep the size limit for the renderer, since the limit on the renderer was already there before I made any changes.

Reply:
Actually let's also remove the size limit on the renderer if you lean towards doing that.

Reply:
I think removing `set_min_inner_size` and `with_min_inner_size` is going to make things significantly easier without really impacting the results at all. So yeah, I think it would probably be smarter to just get rid of it.